runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6105,19 +6104,45 @@ J9::X86::TreeEvaluator::VMmonentEvaluator(
       bool success = TR::TreeEvaluator::monEntryExitHelper(true, node, reservingLock, normalLockPreservingReservation, helper, cg);
       if (success)
          node->setSymbolReference(comp->getSymRefTab()->findOrCreateRuntimeHelper(helper, true, true, true));
+
+      if (reservingLock)
+         {
+         uint32_t reservableLwValue = RES_BIT;
+         if (TR::Options::_aggressiveLockReservation)
+            reservableLwValue = 0;
+
+         TR::Node *reservableLwNode = NULL;
+         if (TR::Compiler->target.is32Bit() || fej9->generateCompressedLockWord())
+            reservableLwNode = TR::Node::iconst(node, reservableLwValue);
+         else
+            reservableLwNode = TR::Node::lconst(node, reservableLwValue);
+
```

Comment:
This way it's the same size as the lock word. Otherwise, on 64-bit with an uncompressed lock word, the helper would have to either zero-extend the value, or rely on the caller having done so even though it's only passing a 32-bit value. I suppose the calling convention might specify that the high half of a register used to pass a 32-bit integer must be zero, but at least GCC doesn't seem to think that's the case on my system.

Reply:
Sounds like a reasonable comment to add to the code as it is definitely non-obvious.